prometheus_es_exporter/utils.py

```diff
@@ -52,7 +52,8 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-def nice_shutdown(shutdown_signals=(signal.SIGINT, signal.SIGTERM)):
+def nice_shutdown(executor=None,
+                  shutdown_signals=(signal.SIGINT, signal.SIGTERM)):
```

Comment:
I might be missing something, but I don't think we actually have access to `executor` to pass in when we call `@nice_shutdown`? At least without more significant restructuring...

I'm not sure it's worth it TBH - we don't wait for the current executing query to finish currently, so don't think it matters if we don't when using threads either. The query result won't be able to be used anyway.

Easier to just leave `nice_shutdown()` as is?

Reply:
Yeah, that seems reasonable. I think I goofed up when pushing this change, and meant to add it to the @nice_shutdown call in __init__.py, but you're right that it works just fine without.